### DISCUSSION TITLE
DT showing components as outdated with hint to incompatible newer version

### DISCUSSION THREAD

**Discussion by logicaloud** [DependencyTrack/dependency-track#2948] [General]
Hi, I'm experimenting with DT and noticed that DT marks components as outdated when newer, possibly incompatible versions are available. In this instance, the application is .NET 6 based and a component is marked as outdated because there is a newer .NET 7 and .NET 8 Preview version available. For example:

![image](https://github.com/DependencyTrack/dependency-track/assets/47704763/d49a044e-4b8e-40b2-ab45-6bb07cd27700)

As a result, I see many components marked as outdated when in fact the latest .NET 6 version is installed.

I'm not familiar with DT; is there some configuration that would allow not flagging components as outdated when the newer component has a different major version number? I.e. per project? Would that kind of approach have any pitfalls?

Any other advise for dealing with false positives in this scenario?

**Comment by stevespringett** [comment]
DT doesn't currently account for compatibility. Doing so would be a nice improvement, but not at the expense of not informing users they are using outdated components. Figuring out a way to represent both would be ideal.

Is there any public source of compatibility information that DT could leverage to know what library versions may introduce incompatibilities? I'm not aware of any, but if they exist, we could potentially add support for them in a future version of DT.

**Comment by logicaloud** [comment]
Thank you for the clarification. For any .NET project almost all components would currently be flagged as outdated unless the latest .NET 8 "preview" version is used. Semantic versioning may give a clue here but is probably not generally reliable, especially with the number of different software platforms in mind. I'm not aware of any compatibility source of information. 

Maybe some form of version check policy could be supported as a configuration option per project? Then users could define some logical / regular expression that is applied for the "outdated" checks to exclude, for example, component versions that have a different major version number or versions that contain "preview"/"alpha"/"beta/"dev"", or similar. Would that work?
